src/devices/bus/spectrum/mface.cpp

```diff
@@ -24,37 +97,77 @@ DEFINE_DEVICE_TYPE(SPECTRUM_MPRINT, spectrum_mprint_device, "spectrum_mprint", "
 //  INPUT_PORTS( mface )
 //-------------------------------------------------
 
-INPUT_PORTS_START(mface)
+INPUT_PORTS_START( mface )
 	PORT_START("BUTTON")
-	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Multiface") PORT_CODE(KEYCODE_F12) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface1_device, magic_button, 0)
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("Red Button") PORT_CODE(KEYCODE_BACKSPACE) PORT_CHANGED_MEMBER(DEVICE_SELF, spectrum_mface_base_device, magic_button, 0)
+INPUT_PORTS_END
+
+INPUT_PORTS_START( mface1 )
+	PORT_INCLUDE( mface )
+	
+	PORT_START("CONFIG")
+	PORT_CONFNAME(0x01, 0x00, "Disable Switch")
```

Comment:
Right now if the user hits the "magic button" and then quickly flips this off before NMI is cleared, NMI will be stuck on.  You need to either always clear NMI when the acknowledge address it read, or add a port changed member on this switch to drop NMI if the interface is disabled while NMI is asserted.

Reply:
I was aware of this but I think it's actually the correct behaviour of the h/w. The multiface osd doesn't clear the nmi until it is closed with the 'r' key. So if it's on screen and the disable switch it opened then the nmi is never cleared. The circuit is a little odd, the "magic button" triggers a flip-flop that immediately disables the button, the disable switch breaks a signal made up of /rd and /iorq lines, I think the original designers just assumed you wouldn't do that (why would you?). I think I'll just remove the disable switch for now until I can confirm exact behaviour on real h/w.

Reply:
Oh, if the real hardware actually does let the NMI get stuck like that, just add a note about it.  I wouldn't have expected the real hardware to do that - I would have thought it would do the "safe" thing and either still allow the NMI to be disabled or drop it on being disabled.